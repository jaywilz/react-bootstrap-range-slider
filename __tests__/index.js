import React from 'react';
import renderer from 'react-test-renderer'

import RangeSlider from '../src/index';

describe('RangeSlider', () => {

  it('should render correctly when only required props provided', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when tooltip = off', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='off'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when size=sm provided', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='off'
          size='sm'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when size=lg provided', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='off'
          size='lg'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when disabled', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='off'
          disabled
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when +ve min provided', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='off'
          min={10}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when -ve min provided', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='off'
          min={-10}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when max provided', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='off'
          max={20}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when min and max provided', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='off'
          min={10}
          max={20}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when step provided', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='off'
          step={25}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when variant provided', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='off'
          variant='secondary'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should pass inputProps to input', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='off'
          inputProps={{'data-a-prop':'test'}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when tooltip = auto', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='auto'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when tooltip explicitly = auto', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='auto'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when tooltipPlacement = top', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltipPlacement='top'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly when tooltipPlacement explicitly = bottom', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltipPlacement='bottom'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly with tooltip label', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltipLabel={value => `${value} km`}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should pass tooltipProps to tooltip div', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltipProps={{'data-a-tooltip-prop':'test'}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should pass tooltipStyle to tooltip div', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltipStyle={{'background':'blue'}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should render correctly with bsPrefix', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={25}
          onChange={() => {}}
          tooltip='off'
          bsPrefix='a-prefix'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should correcly calculate tooltip left position at 0', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={0}
          onChange={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should correcly calculate tooltip left position at 50%', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={50}
          onChange={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should correcly calculate tooltip left position at 100%', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={100}
          onChange={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should correcly calculate tooltip left position at 0 when size = sm', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={0}
          onChange={() => {}}
          size='sm'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should correcly calculate tooltip left position at 50% when size = sm', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={50}
          onChange={() => {}}
          size='sm'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should correcly calculate tooltip left position at 100% when size = sm', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={100}
          onChange={() => {}}
          size='sm'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should correcly calculate tooltip left position at 0 when size = lg', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={0}
          onChange={() => {}}
          size='lg'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should correcly calculate tooltip left position at 50% when size = lg', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={50}
          onChange={() => {}}
          size='lg'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should correcly calculate tooltip left position at 100% when size = lg', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={100}
          onChange={() => {}}
          size='lg'
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should correcly calculate tooltip left position at 0 when min and max provided', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={10}
          onChange={() => {}}
          min={10}
          max={20}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should correcly calculate tooltip left position at 50% when min and max provided', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={15}
          onChange={() => {}}
          min={10}
          max={20}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

  it('should correcly calculate tooltip left position at 100% when min and max provided', () => {

    const tree = renderer
      .create(
        <RangeSlider
          value={20}
          onChange={() => {}}
          min={10}
          max={20}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

  });

});
