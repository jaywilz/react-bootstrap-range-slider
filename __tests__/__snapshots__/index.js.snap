// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider should correcly calculate tooltip left position at 0 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={0}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(0% + 10px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      0
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should correcly calculate tooltip left position at 0 when min and max provided 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={20}
    aria-valuemin={10}
    aria-valuenow={10}
    className="range-slider range-slider--primary"
    disabled={false}
    max={20}
    min={10}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={10}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(0% + 10px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      10
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should correcly calculate tooltip left position at 0 when size = lg 1`] = `
<span
  className="range-slider__wrap range-slider__wrap--lg"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="range-slider range-slider--lg range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={0}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(0% + 12px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      0
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should correcly calculate tooltip left position at 0 when size = sm 1`] = `
<span
  className="range-slider__wrap range-slider__wrap--sm"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="range-slider range-slider--sm range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={0}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(0% + 8px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      0
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should correcly calculate tooltip left position at 50% 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={50}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={50}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(50% + 0px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      50
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should correcly calculate tooltip left position at 50% when min and max provided 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={20}
    aria-valuemin={10}
    aria-valuenow={15}
    className="range-slider range-slider--primary"
    disabled={false}
    max={20}
    min={10}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={15}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(50% + 0px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      15
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should correcly calculate tooltip left position at 50% when size = lg 1`] = `
<span
  className="range-slider__wrap range-slider__wrap--lg"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={50}
    className="range-slider range-slider--lg range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={50}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(50% + 0px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      50
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should correcly calculate tooltip left position at 50% when size = sm 1`] = `
<span
  className="range-slider__wrap range-slider__wrap--sm"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={50}
    className="range-slider range-slider--sm range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={50}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(50% + 0px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      50
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should correcly calculate tooltip left position at 100% 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={100}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={100}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(100% + -10px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      100
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should correcly calculate tooltip left position at 100% when min and max provided 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={20}
    aria-valuemin={10}
    aria-valuenow={20}
    className="range-slider range-slider--primary"
    disabled={false}
    max={20}
    min={10}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={20}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(100% + -10px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      20
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should correcly calculate tooltip left position at 100% when size = lg 1`] = `
<span
  className="range-slider__wrap range-slider__wrap--lg"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={100}
    className="range-slider range-slider--lg range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={100}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(100% + -12px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      100
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should correcly calculate tooltip left position at 100% when size = sm 1`] = `
<span
  className="range-slider__wrap range-slider__wrap--sm"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={100}
    className="range-slider range-slider--sm range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={100}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(100% + -8px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      100
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should pass inputProps to input 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    data-a-prop="test"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
</span>
`;

exports[`RangeSlider should pass tooltipProps to tooltip div 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    data-a-tooltip-prop="test"
    style={
      Object {
        "left": "calc(25% + 5px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      25
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should pass tooltipStyle to tooltip div 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "background": "blue",
        "left": "calc(25% + 5px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      25
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should render correctly when +ve min provided 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={10}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={10}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
</span>
`;

exports[`RangeSlider should render correctly when -ve min provided 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={-10}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={-10}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
</span>
`;

exports[`RangeSlider should render correctly when disabled 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider disabled range-slider--primary"
    disabled={true}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
</span>
`;

exports[`RangeSlider should render correctly when max provided 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={20}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={20}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
</span>
`;

exports[`RangeSlider should render correctly when min and max provided 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={20}
    aria-valuemin={10}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={20}
    min={10}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
</span>
`;

exports[`RangeSlider should render correctly when only required props provided 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(25% + 5px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      25
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should render correctly when size=lg provided 1`] = `
<span
  className="range-slider__wrap range-slider__wrap--lg"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--lg range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
</span>
`;

exports[`RangeSlider should render correctly when size=sm provided 1`] = `
<span
  className="range-slider__wrap range-slider__wrap--sm"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--sm range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
</span>
`;

exports[`RangeSlider should render correctly when step provided 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    step={25}
    type="range"
    value={25}
  />
</span>
`;

exports[`RangeSlider should render correctly when tooltip = auto 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(25% + 5px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      25
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should render correctly when tooltip = off 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
</span>
`;

exports[`RangeSlider should render correctly when tooltip explicitly = auto 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(25% + 5px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      25
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should render correctly when tooltipPlacement = top 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--top"
    style={
      Object {
        "left": "calc(25% + 5px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      25
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should render correctly when tooltipPlacement explicitly = bottom 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(25% + 5px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      25
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;

exports[`RangeSlider should render correctly when variant provided 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--secondary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
</span>
`;

exports[`RangeSlider should render correctly with bsPrefix 1`] = `
<span
  className="a-prefix__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="a-prefix a-prefix--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
</span>
`;

exports[`RangeSlider should render correctly with tooltip label 1`] = `
<span
  className="range-slider__wrap"
>
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="range-slider range-slider--primary"
    disabled={false}
    max={100}
    min={0}
    onChange={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    type="range"
    value={25}
  />
  <div
    className="range-slider__tooltip range-slider__tooltip--auto range-slider__tooltip--bottom"
    style={
      Object {
        "left": "calc(25% + 5px)",
      }
    }
  >
    <div
      className="range-slider__tooltip__label"
    >
      25 km
    </div>
    <div
      className="range-slider__tooltip__caret"
    />
  </div>
</span>
`;
